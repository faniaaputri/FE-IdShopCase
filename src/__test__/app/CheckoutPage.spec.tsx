import CheckOutPage from "@/app/(customer)/checkout/page";
import Home from "@/app/page";
import { render } from "@testing-library/react";

describe("CheckoutPage", () => {
  it("should render", () => {
    const page = render(<CheckOutPage />);
    expect(page).toMatchSnapshot();
  });
});
