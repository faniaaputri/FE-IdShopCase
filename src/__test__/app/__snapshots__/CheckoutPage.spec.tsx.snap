// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`CheckoutPage should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full h-full flex flex-col gap-3 py-2"
      >
        <div
          class="w-full h-[30%] bg border rounded-xs grid grid-cols-6 p-10"
        >
          <p
            class="text-lg font-semibold text-foreground col-span-2"
          >
            Pengiriman
          </p>
          <div
            class="col-span-3"
          >
            <p
              class="text-lg font-medium text-foreground"
            >
              Regular
            </p>
            <div
              class="font-light text-xs"
            >
              <p>
                Garansi Tiba : 27 - 29 Jul
              </p>
              <p>
                Voucher s/d Rp10.000 jika pesanan belum tiba 29 Jul 2025.
              </p>
            </div>
          </div>
        </div>
        <div
          class="w-full h-[10%] bg border rounded-xs grid grid-cols-6 px-10 py-1"
        >
          <p
            class="text-lg font-semibold text-foreground col-span-2"
          >
            Alamat Pengiriman
          </p>
          <div
            class="col-span-3"
          >
            <p
              class="text-lg font-medium text-foreground"
            >
              Regular
            </p>
            <div
              class="font-light text-xs"
            >
              <p>
                Jalan Jend Sudirman, Rt 14/Rw 1, Bendungan Hilir, G, KAB. BOGOR - CIAMPEA, JAWA BARAT, ID 16623
              </p>
            </div>
          </div>
          <div
            class="flex justify-center"
          >
            <button
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive text-primary underline-offset-4 hover:underline h-9 px-4 py-2 has-[>svg]:px-3 font-semibold text-lg"
              data-slot="button"
            >
              Ubah
            </button>
          </div>
        </div>
        <div
          class="w-full h-[30%] bg border rounded-xs grid grid-cols-4 px-10 py-12"
        >
          <div
            class="col-span-3 flex flex-row gap-4"
          >
            <div
              class="relative w-[12%] h-full"
            >
              <img
                alt="checkout-image"
                class="object-cover object-center"
                data-nimg="fill"
                decoding="async"
                loading="lazy"
                sizes="100vw"
                src="/_next/image?url=%2Fimages%2Fproduct-2.jpeg&w=3840&q=75"
                srcset="/_next/image?url=%2Fimages%2Fproduct-2.jpeg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fproduct-2.jpeg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fproduct-2.jpeg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fproduct-2.jpeg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fproduct-2.jpeg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fproduct-2.jpeg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fproduct-2.jpeg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fproduct-2.jpeg&w=3840&q=75 3840w"
                style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
              />
            </div>
            <div
              class="h-full flex flex-col justify-between"
            >
              <div>
                <h2
                  class="text-2xl font-semibold text-foreground"
                >
                  case iphone 16 pro max custom
                </h2>
                <p
                  class="text-xs font-light"
                >
                  Material Diamond Impact
                </p>
              </div>
              <div>
                <p
                  class="text-xl font-semibold text-foreground"
                >
                  Rp 120000
                </p>
              </div>
            </div>
          </div>
          <div
            class="flex flex-row items-center"
          >
            <p
              class="text-app-semibold-sm"
            >
              Jumlah
            </p>
            <div
              class=" flex-row-center  px-10"
            >
              <div
                class="grid grid-cols-4"
              >
                <button
                  class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 h-9 px-4 py-2 has-[>svg]:px-3 col-span-1 text-foreground rounded-none border-foreground"
                  data-slot="button"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M96 235h320v42H96z"
                    />
                  </svg>
                </button>
                <div
                  class="col-span-2 border border-foreground flex-row-center "
                >
                  <p>
                    1
                  </p>
                </div>
                <button
                  class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 h-9 px-4 py-2 has-[>svg]:px-3 col-span-1 text-foreground rounded-none border-foreground"
                  data-slot="button"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M416 277.333H277.333V416h-42.666V277.333H96v-42.666h138.667V96h42.666v138.667H416v42.666z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
          <div>
            <p />
          </div>
        </div>
        <div
          class="w-full h-[30%] bg border rounded-xs flex flex-col"
        >
          <div
            class="h-2/6 w-full px-10 flex flex-row justify-between items-center"
          >
            <p
              class="text-lg font-semibold text-foreground col-span-2"
            >
              Pengiriman
            </p>
            <div
              class="flex flex-row items-center"
            >
              <p
                class="text-lg font-light text-foreground"
              >
                PAYPAL
              </p>
              <button
                class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive text-primary underline-offset-4 hover:underline h-9 px-4 py-2 has-[>svg]:px-3 font-semibold text-lg"
                data-slot="button"
              >
                Ubah
              </button>
            </div>
          </div>
          <div
            class="bg-border shrink-0 data-[orientation=horizontal]:h-px data-[orientation=horizontal]:w-full data-[orientation=vertical]:h-full data-[orientation=vertical]:w-px"
            data-orientation="horizontal"
            data-slot="separator"
            role="none"
          />
          <div
            class="flex justify-end items-center flex-1 px-10"
          >
            <div
              class="flex flex-col gap-2 w-1/5"
            >
              <div>
                <span>
                  Total Pembayaran :
                </span>
                <span
                  class="text-lg font-bold"
                >
                   Rp 120000
                </span>
              </div>
              <button
                class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary hover:bg-primary/90 h-9 has-[>svg]:px-3 text-background rounded-none py-7 px-10 border-foreground"
                data-slot="button"
              >
                <span
                  class="mx-2"
                >
                  Buat Pesanan
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full h-full flex flex-col gap-3 py-2"
    >
      <div
        class="w-full h-[30%] bg border rounded-xs grid grid-cols-6 p-10"
      >
        <p
          class="text-lg font-semibold text-foreground col-span-2"
        >
          Pengiriman
        </p>
        <div
          class="col-span-3"
        >
          <p
            class="text-lg font-medium text-foreground"
          >
            Regular
          </p>
          <div
            class="font-light text-xs"
          >
            <p>
              Garansi Tiba : 27 - 29 Jul
            </p>
            <p>
              Voucher s/d Rp10.000 jika pesanan belum tiba 29 Jul 2025.
            </p>
          </div>
        </div>
      </div>
      <div
        class="w-full h-[10%] bg border rounded-xs grid grid-cols-6 px-10 py-1"
      >
        <p
          class="text-lg font-semibold text-foreground col-span-2"
        >
          Alamat Pengiriman
        </p>
        <div
          class="col-span-3"
        >
          <p
            class="text-lg font-medium text-foreground"
          >
            Regular
          </p>
          <div
            class="font-light text-xs"
          >
            <p>
              Jalan Jend Sudirman, Rt 14/Rw 1, Bendungan Hilir, G, KAB. BOGOR - CIAMPEA, JAWA BARAT, ID 16623
            </p>
          </div>
        </div>
        <div
          class="flex justify-center"
        >
          <button
            class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive text-primary underline-offset-4 hover:underline h-9 px-4 py-2 has-[>svg]:px-3 font-semibold text-lg"
            data-slot="button"
          >
            Ubah
          </button>
        </div>
      </div>
      <div
        class="w-full h-[30%] bg border rounded-xs grid grid-cols-4 px-10 py-12"
      >
        <div
          class="col-span-3 flex flex-row gap-4"
        >
          <div
            class="relative w-[12%] h-full"
          >
            <img
              alt="checkout-image"
              class="object-cover object-center"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=%2Fimages%2Fproduct-2.jpeg&w=3840&q=75"
              srcset="/_next/image?url=%2Fimages%2Fproduct-2.jpeg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fproduct-2.jpeg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fproduct-2.jpeg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fproduct-2.jpeg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fproduct-2.jpeg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fproduct-2.jpeg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fproduct-2.jpeg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fproduct-2.jpeg&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
          <div
            class="h-full flex flex-col justify-between"
          >
            <div>
              <h2
                class="text-2xl font-semibold text-foreground"
              >
                case iphone 16 pro max custom
              </h2>
              <p
                class="text-xs font-light"
              >
                Material Diamond Impact
              </p>
            </div>
            <div>
              <p
                class="text-xl font-semibold text-foreground"
              >
                Rp 120000
              </p>
            </div>
          </div>
        </div>
        <div
          class="flex flex-row items-center"
        >
          <p
            class="text-app-semibold-sm"
          >
            Jumlah
          </p>
          <div
            class=" flex-row-center  px-10"
          >
            <div
              class="grid grid-cols-4"
            >
              <button
                class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 h-9 px-4 py-2 has-[>svg]:px-3 col-span-1 text-foreground rounded-none border-foreground"
                data-slot="button"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M96 235h320v42H96z"
                  />
                </svg>
              </button>
              <div
                class="col-span-2 border border-foreground flex-row-center "
              >
                <p>
                  1
                </p>
              </div>
              <button
                class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 h-9 px-4 py-2 has-[>svg]:px-3 col-span-1 text-foreground rounded-none border-foreground"
                data-slot="button"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M416 277.333H277.333V416h-42.666V277.333H96v-42.666h138.667V96h42.666v138.667H416v42.666z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div>
          <p />
        </div>
      </div>
      <div
        class="w-full h-[30%] bg border rounded-xs flex flex-col"
      >
        <div
          class="h-2/6 w-full px-10 flex flex-row justify-between items-center"
        >
          <p
            class="text-lg font-semibold text-foreground col-span-2"
          >
            Pengiriman
          </p>
          <div
            class="flex flex-row items-center"
          >
            <p
              class="text-lg font-light text-foreground"
            >
              PAYPAL
            </p>
            <button
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive text-primary underline-offset-4 hover:underline h-9 px-4 py-2 has-[>svg]:px-3 font-semibold text-lg"
              data-slot="button"
            >
              Ubah
            </button>
          </div>
        </div>
        <div
          class="bg-border shrink-0 data-[orientation=horizontal]:h-px data-[orientation=horizontal]:w-full data-[orientation=vertical]:h-full data-[orientation=vertical]:w-px"
          data-orientation="horizontal"
          data-slot="separator"
          role="none"
        />
        <div
          class="flex justify-end items-center flex-1 px-10"
        >
          <div
            class="flex flex-col gap-2 w-1/5"
          >
            <div>
              <span>
                Total Pembayaran :
              </span>
              <span
                class="text-lg font-bold"
              >
                 Rp 120000
              </span>
            </div>
            <button
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary hover:bg-primary/90 h-9 has-[>svg]:px-3 text-background rounded-none py-7 px-10 border-foreground"
              data-slot="button"
            >
              <span
                class="mx-2"
              >
                Buat Pesanan
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
